lib/cc/kafka/consumer.rb

```diff
@@ -22,15 +22,28 @@ def initialize(client_id, seed_brokers, topic, partition)
         )
       end
 
+      def on_start(&block)
+        @on_start = block
+      end
+
+      def on_stop(&block)
+        @on_stop = block
+      end
+
       def on_message(&block)
         @on_message = block
       end
 
       def start
-        trap(:TERM) { stop }
+        trap(:TERM) do
+          @on_stop.call if @on_stop
+          stop
```

Comment:
Does it make sense to work to ensure all `fetch_message` operations are done before you call the `on_stop` callback? Thinking of a potential race condition depending on what we shove into the callback.